tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,283 @@
+"""Test the Logitech Squeezebox config flow."""
+from asynctest import patch
+from pysqueezebox import Server
+
+from homeassistant import config_entries
+from homeassistant.components.squeezebox.const import DOMAIN
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    HTTP_UNAUTHORIZED,
+)
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from tests.common import MockConfigEntry
+
+HOST = "1.1.1.1"
+HOST2 = "2.2.2.2"
+PORT = 9000
+UUID = "test-uuid"
+UNKNOWN_ERROR = "1234"
+
+
+async def mock_discover(_discovery_callback):
+    """Mock discovering a Logitech Media Server."""
+    _discovery_callback(Server(None, HOST, PORT, uuid=UUID))
+
+
+async def mock_failed_discover(_discovery_callback):
+    """Mock unsuccessful discovery by doing nothing."""
+
+
+async def test_user_form(hass):
+    """Test user-initiated flow, including discovery and the edit step."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.SqueezeboxConfigFlow._validate_input",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.squeezebox.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover", mock_discover
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+        assert CONF_HOST in result["data_schema"].schema
+        for key in result["data_schema"].schema:
+            if key == CONF_HOST:
+                assert key.description == {"suggested_value": HOST}
+
+        # test the edit step
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: HOST, CONF_PORT: PORT, CONF_USERNAME: "", CONF_PASSWORD: ""},
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == HOST
+        assert result["data"] == {
+            CONF_HOST: HOST,
+            CONF_PORT: PORT,
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+        }
+
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_form_timeout(hass):
+    """Test we handle server search timeout."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover",
```

Comment:
We shouldn't patch homeassistant code, we should instead patch `pysqueezebox`

Reply:
Got it. This is actually `pysqueezebox` code that I patched after import rather than as `pysqueezebox.async_discover`. Fixing to patch `pysqueezebox` directly.

Reply:
Its fine as-is since the code being patched is `pysqueezebox`

For other reviewers: `from pysqueezebox import Server, async_discover`

Reply:
Actually, I think because I imported it as `from pysqueezebox import async_discover` I have to patch it as `pysqueezebox.components.squeezebox.config_flow.async_discover`